### PR TITLE
Fixing result summary for not existing config file

### PR DESCRIPTION
**Pull request by narasimhan-v**
Result summary beautification was done, which had not considered
non existing config files. Fixing that with this commit.

Signed-off-by: Narasimhan V <sim@linux.vnet.ibm.com>
Reported-by: Satheesh Rajendran <sathnaga@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by narasimhan-v** [comment]
```
# ./avocado-setup.py --run-suite host_iooo
13:05:25 INFO    : Check for environment
13:05:25 INFO    : Creating temporary mux dir
13:05:26 ERROR   : Test Config /root/sim/tests/config/tests//host/iooo.cfg not present
13:05:26 INFO    : Summary of test results can be found below:
TestSuite  TestRun    Summary

13:05:26 INFO    : Results and Configs logged at: /root/sim/tests/results/11-5-2020_13_5_25
13:05:26 INFO    : Removing temporary mux dir
```

```
# ./avocado-setup.py --run-suite host_io_fc_fvt,host_iooo
13:05:49 INFO    : Check for environment
13:05:49 INFO    : Creating temporary mux dir
13:05:51 ERROR   : Test Config /root/sim/tests/config/tests//host/iooo.cfg not present
13:05:51 INFO    : 
13:05:51 INFO    : Running Host Tests Suite io_fc_fvt_fs_mark_fs_mark
13:05:51 INFO    : Running: /usr/local/bin/avocado run avocado-misc-tests/io/disk/fs_mark.py -m /root/sim/tests/tests/avocado-misc-tests/io/disk/fs_mark.py.data/fs_mark.yaml --force-job-id d7420758058506de09f746fd7ca51bc58cfe06e0  --job-results-dir /root/sim/tests/results
/usr/local/lib/python3.6/site-packages/avocado_framework-78.0-py3.6.egg/avocado/plugins/run.py:287: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast, keep-tmp and ignore-missing-references.
  FutureWarning)
Error running method "pre_tests" of plugin "fetchasset": 'Attribute' object has no attribute 'id'
JOB ID     : d7420758058506de09f746fd7ca51bc58cfe06e0
JOB LOG    : /root/sim/tests/results/job-2020-05-11T13.05-d742075/job.log
 (1/3) avocado-misc-tests/io/disk/fs_mark.py:FSMark.test;dir-disk-ext4-1d3d: ERROR: 'FSMark' object has no attribute 'link' (1.27 s)
 (2/3) avocado-misc-tests/io/disk/fs_mark.py:FSMark.test;dir-disk-xfs-8b3d: ERROR: 'FSMark' object has no attribute 'link' (1.24 s)
 (3/3) avocado-misc-tests/io/disk/fs_mark.py:FSMark.test;dir-disk-btrfs-209e: CANCEL: btrfs is not supported with                                 RHEL 7.4 onwards (0.75 s)
RESULTS    : PASS 0 | ERROR 2 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 1
JOB TIME   : 3.55 s
JOB HTML   : /root/sim/tests/results/job-2020-05-11T13.05-d742075/results.html
13:05:55 INFO    : 
13:05:55 INFO    : 
13:05:55 INFO    : Running Host Tests Suite io_fc_fvt_fs_mark_fs_mark.2
13:05:55 INFO    : Running: /usr/local/bin/avocado run avocado-misc-tests/io/disk/fs_mark.py -m /root/sim/tests/tests/avocado-misc-tests/io/disk/fs_mark.py.data/fs_mark.yaml --force-job-id c196823b9bc52330b11e971bb05edc758bedb028  --job-results-dir /root/sim/tests/results
/usr/local/lib/python3.6/site-packages/avocado_framework-78.0-py3.6.egg/avocado/plugins/run.py:287: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast, keep-tmp and ignore-missing-references.
  FutureWarning)
Error running method "pre_tests" of plugin "fetchasset": 'Attribute' object has no attribute 'id'
JOB ID     : c196823b9bc52330b11e971bb05edc758bedb028
JOB LOG    : /root/sim/tests/results/job-2020-05-11T13.05-c196823/job.log
 (1/3) avocado-misc-tests/io/disk/fs_mark.py:FSMark.test;dir-disk-ext4-1d3d: ERROR: 'FSMark' object has no attribute 'link' (1.29 s)
 (2/3) avocado-misc-tests/io/disk/fs_mark.py:FSMark.test;dir-disk-xfs-8b3d: ERROR: 'FSMark' object has no attribute 'link' (1.25 s)
 (3/3) avocado-misc-tests/io/disk/fs_mark.py:FSMark.test;dir-disk-btrfs-209e: CANCEL: btrfs is not supported with                                 RHEL 7.4 onwards (0.77 s)
RESULTS    : PASS 0 | ERROR 2 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 1
JOB TIME   : 3.56 s
JOB HTML   : /root/sim/tests/results/job-2020-05-11T13.05-c196823/results.html
13:06:00 INFO    : 
13:06:00 INFO    : Summary of test results can be found below:
TestSuite                                                                                TestRun    Summary
 
host_io_fc_fvt_fs_mark_fs_mark                                                           Run        Successfully executed
/root/sim/tests/results/job-2020-05-11T13.05-d742075/job.log
| PASS 0 || CANCEL 1 || ERRORS 2 || FAILURES 0 || SKIP 0 || WARN 0 || INTERRUPT 0 |
 
host_io_fc_fvt_fs_mark_fs_mark.2                                                         Run        Successfully executed
/root/sim/tests/results/job-2020-05-11T13.05-c196823/job.log
| PASS 0 || CANCEL 1 || ERRORS 2 || FAILURES 0 || SKIP 0 || WARN 0 || INTERRUPT 0 |

13:06:00 INFO    : Results and Configs logged at: /root/sim/tests/results/11-5-2020_13_5_49
13:06:00 INFO    : Removing temporary mux dir
```

```
# ./avocado-setup.py --run-suite host_iooo,host_io_fc_fvt
13:06:17 INFO    : Check for environment
13:06:17 INFO    : Creating temporary mux dir
13:06:17 ERROR   : Test Config /root/sim/tests/config/tests//host/iooo.cfg not present
13:06:19 INFO    : 
13:06:19 INFO    : Running Host Tests Suite io_fc_fvt_fs_mark_fs_mark
13:06:19 INFO    : Running: /usr/local/bin/avocado run avocado-misc-tests/io/disk/fs_mark.py -m /root/sim/tests/tests/avocado-misc-tests/io/disk/fs_mark.py.data/fs_mark.yaml --force-job-id 0a49795f86c6c03b4e6442164ce53667a9ffa87b  --job-results-dir /root/sim/tests/results
/usr/local/lib/python3.6/site-packages/avocado_framework-78.0-py3.6.egg/avocado/plugins/run.py:287: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast, keep-tmp and ignore-missing-references.
  FutureWarning)
Error running method "pre_tests" of plugin "fetchasset": 'Attribute' object has no attribute 'id'
JOB ID     : 0a49795f86c6c03b4e6442164ce53667a9ffa87b
JOB LOG    : /root/sim/tests/results/job-2020-05-11T13.06-0a49795/job.log
 (1/3) avocado-misc-tests/io/disk/fs_mark.py:FSMark.test;dir-disk-ext4-1d3d: ERROR: 'FSMark' object has no attribute 'link' (1.22 s)
 (2/3) avocado-misc-tests/io/disk/fs_mark.py:FSMark.test;dir-disk-xfs-8b3d: ERROR: 'FSMark' object has no attribute 'link' (1.25 s)
 (3/3) avocado-misc-tests/io/disk/fs_mark.py:FSMark.test;dir-disk-btrfs-209e: CANCEL: btrfs is not supported with                                 RHEL 7.4 onwards (0.77 s)
RESULTS    : PASS 0 | ERROR 2 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 1
JOB TIME   : 3.62 s
JOB HTML   : /root/sim/tests/results/job-2020-05-11T13.06-0a49795/results.html
13:06:23 INFO    : 
13:06:23 INFO    : 
13:06:23 INFO    : Running Host Tests Suite io_fc_fvt_fs_mark_fs_mark.2
13:06:23 INFO    : Running: /usr/local/bin/avocado run avocado-misc-tests/io/disk/fs_mark.py -m /root/sim/tests/tests/avocado-misc-tests/io/disk/fs_mark.py.data/fs_mark.yaml --force-job-id 23598e4518372187411ab3d3f6a6bbee8b0b3e27  --job-results-dir /root/sim/tests/results
/usr/local/lib/python3.6/site-packages/avocado_framework-78.0-py3.6.egg/avocado/plugins/run.py:287: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast, keep-tmp and ignore-missing-references.
  FutureWarning)
Error running method "pre_tests" of plugin "fetchasset": 'Attribute' object has no attribute 'id'
JOB ID     : 23598e4518372187411ab3d3f6a6bbee8b0b3e27
JOB LOG    : /root/sim/tests/results/job-2020-05-11T13.06-23598e4/job.log
 (1/3) avocado-misc-tests/io/disk/fs_mark.py:FSMark.test;dir-disk-ext4-1d3d: ERROR: 'FSMark' object has no attribute 'link' (1.24 s)
 (2/3) avocado-misc-tests/io/disk/fs_mark.py:FSMark.test;dir-disk-xfs-8b3d: ERROR: 'FSMark' object has no attribute 'link' (1.56 s)
 (3/3) avocado-misc-tests/io/disk/fs_mark.py:FSMark.test;dir-disk-btrfs-209e: CANCEL: btrfs is not supported with                                 RHEL 7.4 onwards (0.75 s)
RESULTS    : PASS 0 | ERROR 2 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 1
13:06:28 INFO    : Summary of test results can be found below:
TestSuite                                                                                TestRun    Summary
 
host_io_fc_fvt_fs_mark_fs_mark                                                           Run        Successfully executed
/root/sim/tests/results/job-2020-05-11T13.06-0a49795/job.log
| PASS 0 || CANCEL 1 || ERRORS 2 || FAILURES 0 || SKIP 0 || WARN 0 || INTERRUPT 0 |
 
host_io_fc_fvt_fs_mark_fs_mark.2                                                         Run        Successfully executed
/root/sim/tests/results/job-2020-05-11T13.06-23598e4/job.log
| PASS 0 || CANCEL 1 || ERRORS 2 || FAILURES 0 || SKIP 0 || WARN 0 || INTERRUPT 0 |

13:06:28 INFO    : Results and Configs logged at: /root/sim/tests/results/11-5-2020_13_6_17
13:06:28 INFO    : Removing temporary mux dir
```


```
# ./avocado-setup.py --run-suite host_iooo,host_io_fc_fvt
13:08:08 INFO    : Check for environment
13:08:08 INFO    : Creating temporary mux dir
13:08:08 ERROR   : Test Config /root/sim/tests/config/tests//host/iooo.cfg not present
13:08:10 INFO    : 
13:08:10 INFO    : Running Host Tests Suite io_fc_fvt_fs_mark_fs_mark
13:08:10 INFO    : Running: /usr/local/bin/avocado run avocado-misc-tests/io/disk/fs_mark.py -m /root/sim/tests/tests/avocado-misc-tests/io/disk/fs_mark.py.data/fs_mark.yaml --force-job-id c65d17c6980214d0d4be1f1c28e5415caa4bb7fc  --job-results-dir /root/sim/tests/results
/usr/local/lib/python3.6/site-packages/avocado_framework-78.0-py3.6.egg/avocado/plugins/run.py:287: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast, keep-tmp and ignore-missing-references.
  FutureWarning)
Error running method "pre_tests" of plugin "fetchasset": 'Attribute' object has no attribute 'id'
JOB ID     : c65d17c6980214d0d4be1f1c28e5415caa4bb7fc
JOB LOG    : /root/sim/tests/results/job-2020-05-11T13.08-c65d17c/job.log
 (1/3) avocado-misc-tests/io/disk/fs_mark.py:FSMark.test;dir-disk-ext4-1d3d: ERROR: 'FSMark' object has no attribute 'link' (1.24 s)
 (2/3) avocado-misc-tests/io/disk/fs_mark.py:FSMark.test;dir-disk-xfs-8b3d: ERROR: 'FSMark' object has no attribute 'link' (1.26 s)
 (3/3) avocado-misc-tests/io/disk/fs_mark.py:FSMark.test;dir-disk-btrfs-209e: CANCEL: btrfs is not supported with                                 RHEL 7.4 onwards (0.73 s)
RESULTS    : PASS 0 | ERROR 2 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 1
JOB TIME   : 3.61 s
JOB HTML   : /root/sim/tests/results/job-2020-05-11T13.08-c65d17c/results.html
13:08:14 INFO    : 
13:08:14 INFO    : Summary of test results can be found below:
TestSuite                                                                                TestRun    Summary
 
host_io_fc_fvt_fs_mark_fs_mark                                                           Run        Successfully executed
/root/sim/tests/results/job-2020-05-11T13.08-c65d17c/job.log
| PASS 0 || CANCEL 1 || ERRORS 2 || FAILURES 0 || SKIP 0 || WARN 0 || INTERRUPT 0 |

13:08:14 INFO    : Results and Configs logged at: /root/sim/tests/results/11-5-2020_13_8_8
13:08:14 INFO    : Removing temporary mux dir


2020-05-11 13:08:10 DEBUG   : [avocado-setup.py:530] avocado-misc-tests/io/disk/fss_mark.py does not exist
```

**Comment by narasimhan-v** [comment]
> ```# python avocado-setup.py --run-suite guest_kcdsj --guest-os Fedora.31.ppc64le --no-download
> 03:50:14 INFO    : Check for environment
> Traceback (most recent call last):
>   File "avocado-setup.py", line 735, in <module>
>     test_name_list.append(Testsuites[Testsuites_list[0]].runlink.split('\n')[1])
> ```

Can you paste the wrapper log please. Or is this the case of config not run ?

**Comment by sathnaga** [comment]
> > ```
> > 03:50:14 INFO    : Check for environment
> > Traceback (most recent call last):
> >   File "avocado-setup.py", line 735, in <module>
> >     test_name_list.append(Testsuites[Testsuites_list[0]].runlink.split('\n')[1])
> > ```
> 
> Can you paste the wrapper log please. Or is this the case of config not run ?

config not present

File: avocado-setup.py [hunk 1]
```diff
@@ -730,9 +730,12 @@ def parse_test_config(test_config_file, avocado_bin, enable_kvm):
         test_name_list = []
         for test_suite in Testsuites_list:
             test_name_list.append(Testsuites[test_suite].name)
-        test_name_list.append(Testsuites[Testsuites_list[0]].runlink.split('\n')[0])
-        test_name_list.append(Testsuites[Testsuites_list[0]].runlink.split('\n')[1])
-        longest_name_length = len((sorted(test_name_list, key=len)[-1])) + 5
+            if Testsuites[test_suite].runlink:
```
Comment (sathnaga) [inline avocado-setup.py:733]: previously  `test_name_list` got appended for only one runlink, irrespective of how many ever testsuites, now it does extend for every testsuite?, any reason for this change?

Comment (narasimhan-v) [reply avocado-setup.py:733]: @sathnaga Reason why we did this:

We are finding the length needed for column 1, which is the longest of:
1. TestSuiteName
2. job.log path
3. result link

Now, 2 and 3 are static for one wrapper run, so we earlier added only the 1st testsuite's data there. Now, we are not sure which testsuite will have link and which ones will not, so handled in a generic way by having all testsuite's data added, which adds more than needed data in the test_suite list, but solves our issues.

Comment (sathnaga) [reply avocado-setup.py:733]: Good, capture this explanation in commit message, which will help us in future...

**Comment by narasimhan-v** [comment]
@sathnaga explained all changes in commit message. Please verify.

**Comment by sathnaga** [comment]
@narasimhan-v have pushed a commit to fix the travis failures, pls rebase this commit.

**Pull request merged**